From: Andrea Bolognani <eof@kiyuko.org>
Date: Sat, 25 Feb 2023 18:42:27 +0100
Subject: Disable passt support

Rejecting the value from the XML parser ensures that all the
code related to passt, which is left untouched, is never
executed.

The error message reported in this case is the same that
virXMLPropEnum() would generate when encountering a value
that doesn't exist in the enum.

Forwarded: not-needed
---
 src/conf/domain_conf.c   | 6 ++++++
 tests/qemuxml2argvtest.c | 1 -
 tests/qemuxml2xmltest.c  | 1 -
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/conf/domain_conf.c b/src/conf/domain_conf.c
index 3179558..2792cd1 100644
--- a/src/conf/domain_conf.c
+++ b/src/conf/domain_conf.c
@@ -9037,6 +9037,12 @@ virDomainNetBackendParseXML(xmlNodePtr node,
                        VIR_XML_PROP_NONZERO, &def->backend.type) < 0) {
         return -1;
     }
+    if (def->backend.type == VIR_DOMAIN_NET_BACKEND_PASST) {
+        virReportError(VIR_ERR_XML_ERROR,
+                       _("Invalid value for attribute '%s' in element '%s': '%s'."),
+                       "type", node->name, "passt");
+        return -1;
+    }
 
     def->backend.logFile = virXMLPropString(node, "logFile");
 
diff --git a/tests/qemuxml2argvtest.c b/tests/qemuxml2argvtest.c
index bc4cea6..c0fdccb 100644
--- a/tests/qemuxml2argvtest.c
+++ b/tests/qemuxml2argvtest.c
@@ -1474,7 +1474,6 @@ mymain(void)
     DO_TEST_NOCAPS("net-user");
     DO_TEST_CAPS_ARCH_LATEST_FULL("net-user", "x86_64", ARG_FLAGS, FLAG_SLIRP_HELPER);
     DO_TEST_NOCAPS("net-user-addr");
-    DO_TEST_CAPS_LATEST("net-user-passt");
     DO_TEST_NOCAPS("net-virtio");
     DO_TEST_NOCAPS("net-virtio-device");
     DO_TEST_NOCAPS("net-virtio-disable-offloads");
diff --git a/tests/qemuxml2xmltest.c b/tests/qemuxml2xmltest.c
index 66e0385..d70dd62 100644
--- a/tests/qemuxml2xmltest.c
+++ b/tests/qemuxml2xmltest.c
@@ -459,7 +459,6 @@ mymain(void)
     DO_TEST_NOCAPS("net-vhostuser");
     DO_TEST_NOCAPS("net-user");
     DO_TEST_NOCAPS("net-user-addr");
-    DO_TEST_NOCAPS("net-user-passt");
     DO_TEST_NOCAPS("net-virtio");
     DO_TEST_NOCAPS("net-virtio-device");
     DO_TEST_NOCAPS("net-virtio-disable-offloads");
